repos/system_upgrade/el7toel8/actors/removeupgradebootentry/libraries/library.py

```diff
@@ -1,14 +1,24 @@
 from leapp.exceptions import StopActorExecutionError
 from leapp.libraries.stdlib import CalledProcessError, api, run
-from leapp.models import BootContent
+from leapp.models import BootContent, SystemFacts
 
 
 def remove_boot_entry():
     # we need to make sure /boot is mounted before trying to remove the boot entry
+
+    facts_msg = api.consume(SystemFacts)
+    facts = next(facts_msg, None)
+    if not facts:
+        raise StopActorExecutionError('Could not identify system firmware',
+            details={'details': 'Actor did not receive SystemFacts message.'})
+
+    mount_points_per_firmware = {
+        'bios': ['/boot'],
+        'efi': ['/boot', '/boot/efi']
+    }
+
     try:
-        run([
-            '/bin/mount', '/boot'
-        ])
+        run(['/bin/mount'] + mount_points_per_firmware[facts.firmware])
```

Comment:
I think this should be rather a for loop, no? Otherwise you try to mount /boot to /boot/efi if I am not mistaken

Reply:
You are not, fixed.